libs/core/langchain_core/pydantic_v2/__init__.py

```diff
@@ -0,0 +1,24 @@
+import warnings
+
+from langchain_core.utils.pydantic import get_pydantic_major_version
+
+## Create namespaces for pydantic v1 and v2.
+# This code must stay at the top of the file before other modules may
+# attempt to import pydantic since it adds pydantic_v1 and pydantic_v2 to sys.modules.
+#
+# This hack is done for the following reasons:
+# * Langchain will attempt to remain compatible with both pydantic v1 and v2 since
+#   both dependencies and dependents may be stuck on either version of v1 or v2.
+# * Creating namespaces for pydantic v1 and v2 should allow us to write code that
+#   unambiguously uses either v1 or v2 API.
+# * This change is easier to roll out and roll back.
+
+
+if get_pydantic_major_version() < 2:
```

Comment:
There should be no side effects on import since the import is triggered by default rather than user action